src/waypoint.jsx

```diff
@@ -328,7 +332,7 @@ export default class Waypoint extends React.Component {
   render() {
     // We need an element that we can locate in the DOM to determine where it is
     // rendered relative to the top of its context.
-    return <span style={{fontSize: 0}} />;
+    return <span ref={this.refElement} style={{fontSize: 0}} />;
```

Comment:
I've never been able to prove from looking at the react source that an anonymous function on `ref` does or does not cause a the virtual DOM to be synced to the real DOM. You cannot inspect `ref` in shouldComponentUpdate which would imply to me that is does not cause it to be synced but I was never able to definitively determine that.

eslint if configured to prohibit anonymous functions in `render` will still complain if it is `ref`, but maybe it is just too strict.